packages/remix-dev/compiler.ts

```diff
@@ -316,6 +316,12 @@ async function buildEverything(
   }
 }
 
+const yarnPnpCompatibilityPlugin = NodeResolvePlugin({
+  extensions: [".ts", ".js"],
+  onResolved: (resolved) =>
+    resolved.includes("node_modules") ? { external: true } : resolved,
+});
```

Comment:
This is just an instantiation with default config according to the `@esbuild-plugins/node-resolve` because it’s being used in two places in this file.

I tried both plugins, and this is the only one that worked for me, hence the change.

I don’t know what to fix in the previous plugin in order to make it work

Reply:
I understand that this works, but I rather would keep using the official Yarn PnP plugin & create a fix upstream instead of running with our own version (and the burden of maintaining that ourselves) tbh.

Could you please create a compact reproduction repo & file an issue upstream?
That way we can first look on what's the problem on their end & maybe we can fix it in our repo with just a dependency update & without any extra maintenance.

Reply:
I’ll take a look over the weekend. (Although I still don’t quite understand how this increases our maintenance burden — the plugin is maintained [here](https://github.com/remorses/esbuild-plugins) and is designed specifically for PnP compatibility.)

Reply:
It's not a big difference, but now we just import the official Yarn PnP plugin and that's it.

With your solution, we have to know the API (& options) of the `@esbuild-plugins/node-resolve` plugin.
If they ever have a breaking change in there, it adds extra maintenance work on our side.

An API-less plugin won't have any impact then.